kafka-client/src/main/java/dev/responsive/kafka/api/stores/ResponsiveStores.java

```diff
@@ -141,15 +142,18 @@ public static <K, V> StoreBuilder<KeyValueStore<K, V>> keyValueStoreBuilder(
    * @param storeSupplier the key-value store supplier
    * @param keySerde the key serde. If null, the default.key.serde config will be used
    * @param valueSerde the value serde. If null, the default.value.serde config will be used
-   * @return a store builder that can be used to build a key-value store with the given options
-   *         that uses Responsive's storage for its backend
+   * @return a store builder that can be used to build a key-value store that is compatible
+   *         with the Responsive framework
```

Comment:
what's the behavior if they specify a non-NONE storage.backend and they pass in a non-responsive store supplier?

Reply:
this is, and should be, allowed -- for example someone might want to use async for all their stateful apps, but have one Responsive store and one rocksdb (or other  non-responsive) store.

Note that at the moment, you need to use the ResponsiveStoreBuilder for async, but not necessarily a Responsive store. Hence this change.

But I'm working on removing this requirement (in fact I already have in my draft PR for KIP-1112). I'll probably scrape that part of the draft PR out into a separate PR so we can get rid of this requirement without waiting for 4.0.